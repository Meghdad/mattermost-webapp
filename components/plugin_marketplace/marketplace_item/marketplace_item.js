// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import {FormattedMessage} from 'react-intl';

import {Link} from 'react-router-dom';

import LoadingWrapper from 'components/widgets/loading/loading_wrapper.tsx';
import PluginIcon from 'components/widgets/icons/plugin_icon.jsx';

import {trackEvent} from 'actions/diagnostics_actions.jsx';
import {localizeMessage} from 'utils/utils';

// UpdateVersion renders the version text in the update details, linking out to release notes if available.
export const UpdateVersion = ({version, releaseNotesUrl}) => {
    if (!releaseNotesUrl) {
        return version;
    }

    return (
        <a
            target='_blank'
            rel='noopener noreferrer'
            href={releaseNotesUrl}
        >
            {version}
        </a>
    );
};

UpdateVersion.propTypes = {
    version: PropTypes.string.isRequired,
    releaseNotesUrl: PropTypes.string,
};

// UpdateDetails renders an inline update prompt for plugins, when available.
export const UpdateDetails = ({availableVersion, releaseNotesUrl, installedVersion, isInstalling, onUpdate}) => {
    if (!installedVersion || installedVersion === availableVersion || isInstalling) {
        return null;
    }

    return (
        <div className={classNames('update')}>
            <FormattedMessage
                id='marketplace_modal.list.update_available'
                defaultMessage='Update available:'
            />
            {' '}
            <UpdateVersion
                version={availableVersion}
                releaseNotesUrl={releaseNotesUrl}
            />
            {' - '}
            <b>
                <a onClick={onUpdate}>
                    <FormattedMessage
                        id='marketplace_modal.list.update'
                        defaultMessage='Update'
                    />
                </a>
            </b>
        </div>
    );
};

UpdateDetails.propTypes = {
    availableVersion: PropTypes.string.isRequired,
    releaseNotesUrl: PropTypes.string,
    installedVersion: PropTypes.string,
    isInstalling: PropTypes.bool.isRequired,
    onUpdate: PropTypes.func.isRequired,
};

export class MarketplaceItem extends React.Component {
    static propTypes = {
        id: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired,
        description: PropTypes.string.isRequired,
        version: PropTypes.string.isRequired,
        downloadUrl: PropTypes.string,
        homepageUrl: PropTypes.string,
        releaseNotesUrl: PropTypes.string,
        iconData: PropTypes.string,
        installedVersion: PropTypes.string.isRequired,
        installing: PropTypes.bool.isRequired,
        error: PropTypes.string,
        actions: PropTypes.shape({
            installPlugin: PropTypes.func.isRequired,
            closeMarketplaceModal: PropTypes.func.isRequired,
        }).isRequired,
    };

    onInstall = () => {
        trackEvent('plugins', 'ui_marketplace_download');
        this.props.actions.installPlugin(this.props.id);
    }

    onUpdate = () => {
        trackEvent('plugins', 'ui_marketplace_download_update');
        this.props.actions.installPlugin(this.props.id);
    }

    onConfigure = () => {
        trackEvent('plugins', 'ui_marketplace_configure');
        this.props.actions.closeMarketplaceModal();
    }

    getItemButton() {
        let actionButton = (
            <FormattedMessage
                id='marketplace_modal.list.Install'
                defaultMessage='Install'
            />
        );
        if (this.props.error) {
            actionButton = (
                <FormattedMessage
                    id='marketplace_modal.list.try_again'
                    defaultMessage='Try Again'
                />
            );
        }

        let button = (
            <button
                onClick={this.onInstall}
                className='btn btn-primary'
                disabled={this.props.installing || this.props.downloadUrl === ''}
            >
                <LoadingWrapper
                    loading={this.props.installing}
                    text={localizeMessage('marketplace_modal.installing', 'Installing...')}
                >
                    {actionButton}
                </LoadingWrapper>

            </button>
        );

        if (this.props.installedVersion !== '' && !this.props.installing && !this.props.error) {
            button = (
                <Link
                    to={'/admin_console/plugins/plugin_' + this.props.id}
                >
                    <button
                        onClick={this.onConfigure}
                        className='btn btn-outline'
                    >
                        <FormattedMessage
                            id='marketplace_modal.list.configure'
                            defaultMessage='Configure'
                        />
                    </button>
                </Link>
            );
        }

        return button;
    }

    render() {
        const ariaLabel = `${this.props.name}, ${this.props.description}`.toLowerCase();
        let versionLabel = `(${this.props.version})`;
        if (this.props.installedVersion !== '') {
            versionLabel = `(${this.props.installedVersion})`;
        }

        let pluginIcon;
        if (this.props.iconData) {
            pluginIcon = (
                <div className='icon__plugin icon__plugin--background'>
                    <img src={this.props.iconData}/>
                </div>
            );
        } else {
            pluginIcon = <PluginIcon className='icon__plugin icon__plugin--background'/>;
        }

        let pluginDetails = (
            <>
                {this.props.name} <span className='light subtitle'>{versionLabel}</span>
                <p className={classNames('more-modal__description', {error_text: this.props.error})}>
                    {this.props.error ? this.props.error : this.props.description}
                </p>
            </>
        );

        if (this.props.homepageUrl) {
            pluginDetails = (
                <a
                    aria-label={ariaLabel}
                    className='style--none more-modal__row--link'
                    target='_blank'
                    rel='noopener noreferrer'
                    href={this.props.homepageUrl}
                >
                    {pluginDetails}
                </a>
            );
        } else {
            pluginDetails = (
                <span
                    aria-label={ariaLabel}
                    className='style--none'
                >
                    {pluginDetails}
                </span>
            );
        }

        return (
            <>
                <div
                    className={classNames('more-modal__row', 'more-modal__row--link', {item_error: this.props.error})}
                    key={this.props.id}
                    id={'marketplace-plugin-' + this.props.id}
                >
                    {pluginIcon}
                    <div className='more-modal__details'>
                        {pluginDetails}
                        <UpdateDetails
                            availableVersion={this.props.version}
                            installedVersion={this.props.installedVersion}
                            releaseNotesUrl={this.props.releaseNotesUrl}
                            isInstalling={this.props.installing}
                            onUpdate={this.onUpdate}
                        />
                    </div>
                    <div className='more-modal__actions'>
                        {this.getItemButton()}
                    </div>
                </div>
            </>
        );
    }
}
