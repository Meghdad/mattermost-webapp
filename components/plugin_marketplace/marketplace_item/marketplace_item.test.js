// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import {mountWithIntl as mount} from 'tests/helpers/intl-test-helper.jsx';

import {MarketplaceItem, UpdateDetails} from './marketplace_item';

describe('components/MarketplaceItem', () => {
    describe('UpdateDetails', () => {
        const baseProps = {
            availableVersion: '0.0.2',
            releaseNotesUrl: 'http://example.com/release',
            installedVersion: '0.0.1',
            isInstalling: false,
            onUpdate: () => {},
        };

        describe('should render nothing', () => {
            it('when no installed version', () => {
                const props = {
                    ...baseProps,
                    installedVersion: '',
                };
                const wrapper = mount(
                    <UpdateDetails {...props}/>
                );

                expect(wrapper.isEmptyRender()).toBe(true);
            });

            it('when installed version matches available version', () => {
                const props = {
                    ...baseProps,
                    installedVersion: baseProps.availableVersion,
                };
                const wrapper = mount(
                    <UpdateDetails {...props}/>
                );

                expect(wrapper.isEmptyRender()).toBe(true);
            });

            it('when installing', () => {
                const props = {
                    ...baseProps,
                    isInstalling: true,
                };
                const wrapper = mount(
                    <UpdateDetails {...props}/>
                );

                expect(wrapper.isEmptyRender()).toBe(true);
            });
        });

        it('should render without release notes url', () => {
            const props = {
                ...baseProps,
                releaseNotesUrl: '',
            };

            const wrapper = mount(
                <UpdateDetails {...props}/>
            );

            expect(wrapper).toMatchSnapshot();
        });

        it('should render without release notes url', () => {
            const wrapper = mount(
                <UpdateDetails {...baseProps}/>
            );

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('MarketplaceItem', () => {
        const baseProps = {
            id: 'id',
            name: 'name',
            description: 'test plugin',
            version: '1.0.0',
            downloadUrl: 'http://example.com/download',
            signatureUrl: 'http://example.com/signature',
            homepageUrl: 'http://example.com',
            installedVersion: '',
            iconUrl: '',
            iconData: 'icon',
            installing: false,
            actions: {
                installPlugin: () => {}, // eslint-disable-line no-empty-function
                closeMarketplaceModal: () => {}, // eslint-disable-line no-empty-function
            },
        };

        test('should render', () => {
            const wrapper = shallow(
                <MarketplaceItem {...baseProps}/>
            );

            expect(wrapper).toMatchSnapshot();
        });

        test('should render with no plugin icon', () => {
            const props = {...baseProps};
            delete props.iconData;

            const wrapper = shallow(
                <MarketplaceItem {...props}/>
            );

            expect(wrapper).toMatchSnapshot();
        });

        test('should render wtih no homepage url', () => {
            const props = {...baseProps};
            delete props.homepageUrl;

            const wrapper = shallow(
                <MarketplaceItem {...props}/>
            );

            expect(wrapper).toMatchSnapshot();
        });

        test('should render with server error', () => {
            const props = {
                ...baseProps,
                error: 'An error occurred.',
            };

            const wrapper = shallow(
                <MarketplaceItem {...props}/>
            );

            expect(wrapper).toMatchSnapshot();
        });

        test('should render installed plugin', () => {
            const props = {
                ...baseProps,
                installedVersion: '1.0.0',
            };

            const wrapper = shallow(
                <MarketplaceItem {...props}/>
            );

            expect(wrapper).toMatchSnapshot();
        });

        test('should render with update available', () => {
            const props = {
                ...baseProps,
                installedVersion: '0.9.9',
            };

            const wrapper = shallow(
                <MarketplaceItem {...props}/>
            );

            expect(wrapper).toMatchSnapshot();
        });

        test('should render with update and release notes available', () => {
            const props = {
                ...baseProps,
                installedVersion: '0.9.9',
                releaseNotesUrl: 'http://example.com/release',
            };

            const wrapper = shallow(
                <MarketplaceItem {...props}/>
            );

            expect(wrapper).toMatchSnapshot();
        });
    });
});
