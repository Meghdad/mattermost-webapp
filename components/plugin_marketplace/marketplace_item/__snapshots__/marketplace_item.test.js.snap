// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/MarketplaceItem MarketplaceItem should render 1`] = `
<Fragment>
  <div
    className="more-modal__row more-modal__row--link"
    id="marketplace-plugin-id"
    key="id"
  >
    <div
      className="icon__plugin icon__plugin--background"
    >
      <img
        src="icon"
      />
    </div>
    <div
      className="more-modal__details"
    >
      <a
        aria-label="name, test plugin"
        className="style--none more-modal__row--link"
        href="http://example.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        name
         
        <span
          className="light subtitle"
        >
          (1.0.0)
        </span>
        <p
          className="more-modal__description"
        >
          test plugin
        </p>
      </a>
      <UpdateDetails
        availableVersion="1.0.0"
        installedVersion=""
        isInstalling={false}
        onUpdate={[Function]}
      />
    </div>
    <div
      className="more-modal__actions"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
      >
        <LoadingWrapper
          loading={false}
          text="Installing..."
        >
          <FormattedMessage
            defaultMessage="Install"
            id="marketplace_modal.list.Install"
            values={Object {}}
          />
        </LoadingWrapper>
      </button>
    </div>
  </div>
</Fragment>
`;

exports[`components/MarketplaceItem MarketplaceItem should render installed plugin 1`] = `
<Fragment>
  <div
    className="more-modal__row more-modal__row--link"
    id="marketplace-plugin-id"
    key="id"
  >
    <div
      className="icon__plugin icon__plugin--background"
    >
      <img
        src="icon"
      />
    </div>
    <div
      className="more-modal__details"
    >
      <a
        aria-label="name, test plugin"
        className="style--none more-modal__row--link"
        href="http://example.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        name
         
        <span
          className="light subtitle"
        >
          (1.0.0)
        </span>
        <p
          className="more-modal__description"
        >
          test plugin
        </p>
      </a>
      <UpdateDetails
        availableVersion="1.0.0"
        installedVersion="1.0.0"
        isInstalling={false}
        onUpdate={[Function]}
      />
    </div>
    <div
      className="more-modal__actions"
    >
      <Link
        to="/admin_console/plugins/plugin_id"
      >
        <button
          className="btn btn-outline"
          onClick={[Function]}
        >
          <FormattedMessage
            defaultMessage="Configure"
            id="marketplace_modal.list.configure"
            values={Object {}}
          />
        </button>
      </Link>
    </div>
  </div>
</Fragment>
`;

exports[`components/MarketplaceItem MarketplaceItem should render when not from the marketplace 1`] = `
<Fragment>
  <div
    className="more-modal__row more-modal__row--link"
    id="marketplace-plugin-id"
    key="id"
  >
    <div
      className="icon__plugin icon__plugin--background"
    >
      <img
        src="icon"
      />
    </div>
    <div
      className="more-modal__details"
    >
      <a
        aria-label="name, test plugin"
        className="style--none more-modal__row--link"
        href="http://example.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        name
         
        <span
          className="light subtitle"
        >
          (1.0.0)
        </span>
        <OverlayTrigger
          defaultOverlayShown={false}
          delayShow={400}
          overlay={
            <Tooltip
              bsClass="tooltip"
              id="plugin-marketplace__local-tooltop"
              placement="right"
            >
              <FormattedMessage
                defaultMessage="This plugin is not listed in the marketplace but was installed manually."
                id="marketplace_modal.list.local.tooltip"
                values={Object {}}
              />
            </Tooltip>
          }
          placement="top"
          trigger={
            Array [
              "hover",
              "focus",
            ]
          }
        >
          <span
            className="tag"
          >
            <FormattedMessage
              defaultMessage="Local"
              id="marketplace_modal.list.local"
              values={Object {}}
            />
          </span>
        </OverlayTrigger>
        <p
          className="more-modal__description"
        >
          test plugin
        </p>
      </a>
      <UpdateDetails
        availableVersion="1.0.0"
        installedVersion=""
        isInstalling={false}
        onUpdate={[Function]}
      />
    </div>
    <div
      className="more-modal__actions"
    >
      <button
        className="btn btn-primary"
        disabled={true}
        onClick={[Function]}
      >
        <LoadingWrapper
          loading={false}
          text="Installing..."
        >
          <FormattedMessage
            defaultMessage="Install"
            id="marketplace_modal.list.Install"
            values={Object {}}
          />
        </LoadingWrapper>
      </button>
    </div>
  </div>
</Fragment>
`;

exports[`components/MarketplaceItem MarketplaceItem should render with no plugin icon 1`] = `
<Fragment>
  <div
    className="more-modal__row more-modal__row--link"
    id="marketplace-plugin-id"
    key="id"
  >
    <PluginIcon
      className="icon__plugin icon__plugin--background"
    />
    <div
      className="more-modal__details"
    >
      <a
        aria-label="name, test plugin"
        className="style--none more-modal__row--link"
        href="http://example.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        name
         
        <span
          className="light subtitle"
        >
          (1.0.0)
        </span>
        <p
          className="more-modal__description"
        >
          test plugin
        </p>
      </a>
      <UpdateDetails
        availableVersion="1.0.0"
        installedVersion=""
        isInstalling={false}
        onUpdate={[Function]}
      />
    </div>
    <div
      className="more-modal__actions"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
      >
        <LoadingWrapper
          loading={false}
          text="Installing..."
        >
          <FormattedMessage
            defaultMessage="Install"
            id="marketplace_modal.list.Install"
            values={Object {}}
          />
        </LoadingWrapper>
      </button>
    </div>
  </div>
</Fragment>
`;

exports[`components/MarketplaceItem MarketplaceItem should render with server error 1`] = `
<Fragment>
  <div
    className="more-modal__row more-modal__row--link item_error"
    id="marketplace-plugin-id"
    key="id"
  >
    <div
      className="icon__plugin icon__plugin--background"
    >
      <img
        src="icon"
      />
    </div>
    <div
      className="more-modal__details"
    >
      <a
        aria-label="name, test plugin"
        className="style--none more-modal__row--link"
        href="http://example.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        name
         
        <span
          className="light subtitle"
        >
          (1.0.0)
        </span>
        <p
          className="more-modal__description error_text"
        >
          An error occurred.
        </p>
      </a>
      <UpdateDetails
        availableVersion="1.0.0"
        installedVersion=""
        isInstalling={false}
        onUpdate={[Function]}
      />
    </div>
    <div
      className="more-modal__actions"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
      >
        <LoadingWrapper
          loading={false}
          text="Installing..."
        >
          <FormattedMessage
            defaultMessage="Try Again"
            id="marketplace_modal.list.try_again"
            values={Object {}}
          />
        </LoadingWrapper>
      </button>
    </div>
  </div>
</Fragment>
`;

exports[`components/MarketplaceItem MarketplaceItem should render with update and release notes available 1`] = `
<Fragment>
  <div
    className="more-modal__row more-modal__row--link"
    id="marketplace-plugin-id"
    key="id"
  >
    <div
      className="icon__plugin icon__plugin--background"
    >
      <img
        src="icon"
      />
    </div>
    <div
      className="more-modal__details"
    >
      <a
        aria-label="name, test plugin"
        className="style--none more-modal__row--link"
        href="http://example.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        name
         
        <span
          className="light subtitle"
        >
          (0.9.9)
        </span>
        <p
          className="more-modal__description"
        >
          test plugin
        </p>
      </a>
      <UpdateDetails
        availableVersion="1.0.0"
        installedVersion="0.9.9"
        isInstalling={false}
        onUpdate={[Function]}
        releaseNotesUrl="http://example.com/release"
      />
    </div>
    <div
      className="more-modal__actions"
    >
      <Link
        to="/admin_console/plugins/plugin_id"
      >
        <button
          className="btn btn-outline"
          onClick={[Function]}
        >
          <FormattedMessage
            defaultMessage="Configure"
            id="marketplace_modal.list.configure"
            values={Object {}}
          />
        </button>
      </Link>
    </div>
  </div>
</Fragment>
`;

exports[`components/MarketplaceItem MarketplaceItem should render with update available 1`] = `
<Fragment>
  <div
    className="more-modal__row more-modal__row--link"
    id="marketplace-plugin-id"
    key="id"
  >
    <div
      className="icon__plugin icon__plugin--background"
    >
      <img
        src="icon"
      />
    </div>
    <div
      className="more-modal__details"
    >
      <a
        aria-label="name, test plugin"
        className="style--none more-modal__row--link"
        href="http://example.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        name
         
        <span
          className="light subtitle"
        >
          (0.9.9)
        </span>
        <p
          className="more-modal__description"
        >
          test plugin
        </p>
      </a>
      <UpdateDetails
        availableVersion="1.0.0"
        installedVersion="0.9.9"
        isInstalling={false}
        onUpdate={[Function]}
      />
    </div>
    <div
      className="more-modal__actions"
    >
      <Link
        to="/admin_console/plugins/plugin_id"
      >
        <button
          className="btn btn-outline"
          onClick={[Function]}
        >
          <FormattedMessage
            defaultMessage="Configure"
            id="marketplace_modal.list.configure"
            values={Object {}}
          />
        </button>
      </Link>
    </div>
  </div>
</Fragment>
`;

exports[`components/MarketplaceItem MarketplaceItem should render wtih no homepage url 1`] = `
<Fragment>
  <div
    className="more-modal__row more-modal__row--link"
    id="marketplace-plugin-id"
    key="id"
  >
    <div
      className="icon__plugin icon__plugin--background"
    >
      <img
        src="icon"
      />
    </div>
    <div
      className="more-modal__details"
    >
      <span
        aria-label="name, test plugin"
        className="style--none"
      >
        name
         
        <span
          className="light subtitle"
        >
          (1.0.0)
        </span>
        <p
          className="more-modal__description"
        >
          test plugin
        </p>
      </span>
      <UpdateDetails
        availableVersion="1.0.0"
        installedVersion=""
        isInstalling={false}
        onUpdate={[Function]}
      />
    </div>
    <div
      className="more-modal__actions"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
      >
        <LoadingWrapper
          loading={false}
          text="Installing..."
        >
          <FormattedMessage
            defaultMessage="Install"
            id="marketplace_modal.list.Install"
            values={Object {}}
          />
        </LoadingWrapper>
      </button>
    </div>
  </div>
</Fragment>
`;

exports[`components/MarketplaceItem UpdateDetails should render without release notes url 1`] = `
<UpdateDetails
  availableVersion="0.0.2"
  installedVersion="0.0.1"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": "Etc/UTC",
    }
  }
  isInstalling={false}
  onUpdate={[Function]}
  releaseNotesUrl=""
>
  <div
    className="update"
  >
    <FormattedMessage
      defaultMessage="Update available:"
      id="marketplace_modal.list.update_available"
      values={Object {}}
    >
      <span>
        Update available:
      </span>
    </FormattedMessage>
     
    <UpdateVersion
      releaseNotesUrl=""
      version="0.0.2"
    >
      0.0.2
    </UpdateVersion>
     - 
    <b>
      <a
        onClick={[Function]}
      >
        <FormattedMessage
          defaultMessage="Update"
          id="marketplace_modal.list.update"
          values={Object {}}
        >
          <span>
            Update
          </span>
        </FormattedMessage>
      </a>
    </b>
  </div>
</UpdateDetails>
`;

exports[`components/MarketplaceItem UpdateDetails should render without release notes url 2`] = `
<UpdateDetails
  availableVersion="0.0.2"
  installedVersion="0.0.1"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": "Etc/UTC",
    }
  }
  isInstalling={false}
  onUpdate={[Function]}
  releaseNotesUrl="http://example.com/release"
>
  <div
    className="update"
  >
    <FormattedMessage
      defaultMessage="Update available:"
      id="marketplace_modal.list.update_available"
      values={Object {}}
    >
      <span>
        Update available:
      </span>
    </FormattedMessage>
     
    <UpdateVersion
      releaseNotesUrl="http://example.com/release"
      version="0.0.2"
    >
      <a
        href="http://example.com/release"
        rel="noopener noreferrer"
        target="_blank"
      >
        0.0.2
      </a>
    </UpdateVersion>
     - 
    <b>
      <a
        onClick={[Function]}
      >
        <FormattedMessage
          defaultMessage="Update"
          id="marketplace_modal.list.update"
          values={Object {}}
        >
          <span>
            Update
          </span>
        </FormattedMessage>
      </a>
    </b>
  </div>
</UpdateDetails>
`;
